x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/MountSearchableSnapshotRequest.java

```diff
@@ -0,0 +1,223 @@
+        this.indexSettings = readSettingsFromStream(in);
+        this.ignoredIndexSettings = in.readStringArray();
+        this.waitForCompletion = in.readBoolean();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        out.writeString(mountedIndexName);
+        out.writeString(repositoryName);
+        out.writeString(snapshotName);
+        out.writeOptionalString(snapshotIndexName);
+        writeSettingsToStream(indexSettings, out);
+        out.writeStringArray(ignoredIndexSettings);
+        out.writeBoolean(waitForCompletion);
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+       return null;
+    }
+
+    /**
+     * @return the name of the index that will be created
+     */
+    public String mountedIndexName() {
+        return mountedIndexName;
+    }
+
+    /**
+     * @return the name of the repository
+     */
+    public String repositoryName() {
+        return this.repositoryName;
+    }
+
+    /**
+     * @return the name of the snapshot.
+     */
+    public String snapshotName() {
+        return this.snapshotName;
+    }
+
+    /**
+     * @return the name of the index contained in the snapshot
+     */
+    public String snapshotIndexName() {
+        return snapshotIndexName;
+    }
+
+    /**
+     * @return true if the operation will wait for completion
+     */
+    public boolean waitForCompletion() {
+        return waitForCompletion;
+    }
+
+    /**
+     * @return settings that should be added to the index when it is mounted
+     */
+    public Settings indexSettings() {
+        return this.indexSettings;
+    }
+
+    /**
+     * @return the names of settings that should be removed from the index when it is mounted
+     */
+    public String[] ignoreIndexSettings() {
+        return ignoredIndexSettings;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(INDEX_FIELD.getPreferredName(), snapshotIndexName);
+        builder.field(RENAMED_INDEX_FIELD.getPreferredName(), mountedIndexName);
+        if (indexSettings.isEmpty() == false) {
+            builder.startObject(INDEX_SETTINGS_FIELD.getPreferredName());
+            indexSettings.toXContent(builder, params);
+            builder.endObject();
+        }
+        if (ignoredIndexSettings.length > 0) {
+            builder.startArray(IGNORE_INDEX_SETTINGS_FIELD.getPreferredName());
+            for (String ignoreIndexSetting : ignoredIndexSettings) {
+                builder.value(ignoreIndexSetting);
+            }
+            builder.endArray();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getDescription() {
+        return "mount snapshot [" + repositoryName + ":" + snapshotName + ":" + snapshotIndexName + "->" + mountedIndexName + "]";
```

Comment:
Maybe:
```suggestion
        return "mount snapshot [" + repositoryName + ":" + snapshotName + ":" + snapshotIndexName + "] as index [" + mountedIndexName + "]";
```

Reply:
++ done in cca756ecba0db5cc514e13568c9413681264254e.